include/envoy/server/resource_monitor_config.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/server/resource_monitor.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+class ResourceMonitorFactoryContext {
+public:
+  virtual ~ResourceMonitorFactoryContext() {}
+
+  /**
+   * @return Event::Dispatcher& the main thread's dispatcher. This dispatcher should be used
+   *         for all singleton processing.
+   */
+  virtual Event::Dispatcher& dispatcher() PURE;
```

Comment:
I would just inline this and skip the context unless you anticipate this growing significantly.

Reply:
Done